src/main/java/com/epam/team1/emotional/help/service/QuizHistoryService.java

```diff
@@ -48,10 +48,29 @@ public QuizHistory createQuizHistory(User user, Result result, Quiz quiz){
                 .completeDate(LocalDateTime.now())
                 .build();
     }
+    public QuizHistory createQuizHistory(User user, Questionnaire questionnaire, Integer pointsScored, Result result){
+        return QuizHistory.builder()
+                .user(user)
+                .questionnaire(questionnaire)
+                .points(pointsScored)
+                .result(result)
+                .completeDate(LocalDateTime.now()).build();
+    }
+
 
     public List<QuizHistoryResponseDto> getQuizHistoryByUser(Long userId) {
         User user = userService.getUserById(userId);
         List<QuizHistory> quizHistories = quizHistoryRepository.findByUser(user);
         return quizHistoryMapper.toQuizHistoryResponseDtoList(quizHistories);
     }
+    public void saveQuizForUnauthenticatedUser(SendQuizResultToEmailDto dto){
+        Optional<User> userOptional= userService.findUserByEmail(dto.getEmail());
+        Questionnaire questionnaire = questionnaireService.getById(dto.getQuestionnaireId());
+        Result result = resultService.getResultByQuestionnaireAndPoints(questionnaire,dto.getPoints());
+
+        if (userOptional.isEmpty()||questionnaire==null) return;
```

Comment:
Try to avoid silent return in such cases. You may throw an exception or at least log it.